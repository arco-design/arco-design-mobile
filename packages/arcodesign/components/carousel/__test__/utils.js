import { act } from 'react-dom/test-utils';
import { createMoveTouchEventObject, createStartTouchEventObject } from '../../../tests/helpers/mockEvent';

export function mockSwipeStart(map, clientXMap) {
    const { touchstart, touchmove } = clientXMap;
    act(() => {
        map.touchstart(createStartTouchEventObject({ x: touchstart, y: 0 }));
        map.touchmove(createMoveTouchEventObject({ x: touchmove, y: 0 }));
    });
}
export function mockSwipeEnd(wrapper, className, touchend) {
    wrapper.find(`.${className}`).simulate('touchEnd', createMoveTouchEventObject({ x: touchend, y: 0 }));
    act(() => {
        jest.advanceTimersByTime(600);
    });
    wrapper.update();
}

export function mockSwipe(map, wrapper, className, clientXMap) {
    const { touchend } = clientXMap;
    mockSwipeStart(map, clientXMap);
    mockSwipeEnd(wrapper, className, touchend);
}
