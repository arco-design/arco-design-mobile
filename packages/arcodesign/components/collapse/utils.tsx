import React, { Context } from 'react';
import { GroupContextParams } from './type';

export const defaultContextKey = '__COLLAPSE_CONTEXT__';
export const allContexts: Record<string, Context<GroupContextParams>> = {};

export function getCollapseContext(key: string) {
    if (!allContexts[key]) {
        allContexts[key] = React.createContext<GroupContextParams>({
            isGroup: false,
            value: [],
            onCollapse: () => {},
        });
    }
    return allContexts[key];
}

export const CollapseKeyContext = React.createContext<{ key: string }>({
    key: defaultContextKey,
});
