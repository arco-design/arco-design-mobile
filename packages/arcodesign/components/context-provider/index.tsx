import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
