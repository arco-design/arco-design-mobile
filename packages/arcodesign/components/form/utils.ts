import { IRules, ValidatorError } from '@arco-design/mobile-utils';

export const isFieldRequired = (rules: IRules[] = []) => {
    return (rules || []).some(rule => rule?.required);
};

export const getErrorAndWarnings = (result: ValidatorError[]) => {
    let errors: string[] = [];
    let warnings: string[] = [];
    result.map(({ message = [], validateLevel = 'error' }) => {
        if (!message?.length) {
            return;
        }
        if (validateLevel === 'warning') {
            warnings = [...warnings, ...message];
        } else {
            errors = [...errors, ...message];
        }
    });
    return { warnings, errors };
};
