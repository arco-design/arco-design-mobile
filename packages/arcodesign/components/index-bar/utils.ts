import { Children, isValidElement, ReactElement, ReactNode } from 'react';
import { IndexBarIndexType } from './type';

export function filterValidIndexBarChild(children: ReactNode) {
    const validChildren: ReactElement[] = [];
    Children.forEach(children, child => {
        if (!isValidElement(child)) {
            console.warn('<IndexBar /> only accept <IndexBar.Group /> as the children');
            return;
        }
        validChildren.push(child);
    });
    return validChildren;
}

// index有可能是number或者string，但是绑在data-index的attr上后
// 通过滑动事件得到的都是string，需要处理下
export function getFormatIndex(
    index: IndexBarIndexType,
    activeIndex: IndexBarIndexType | undefined,
) {
    const activeIndexType = typeof activeIndex;
    const indexType = typeof index;
    if (indexType === activeIndexType) {
        return index;
    }
    if (activeIndexType === 'number') {
        return Number(index);
    }
    return index.toString();
}

export function isValidIndex(index: unknown): index is IndexBarIndexType {
    return typeof index === 'number' || typeof index === 'string';
}

export function getGroupDomFromIndex(containerDom: HTMLDivElement, activeIndex: IndexBarIndexType) {
    let targetDom: HTMLDivElement | null = null;

    const containerChildren = containerDom.children;
    for (let i = 0; i < containerChildren.length; i++) {
        const dom = containerChildren[i] as HTMLDivElement;
        const dataset = dom?.dataset;
        if (dataset && dataset.index) {
            const formatIndex = getFormatIndex(dataset.index, activeIndex);
            if (formatIndex === activeIndex) {
                targetDom = dom;
                break;
            }
        }
    }

    return targetDom;
}
