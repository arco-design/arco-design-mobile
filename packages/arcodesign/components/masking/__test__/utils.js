import React, { forwardRef, useImperativeHandle, useRef, useState } from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Button from '../../button';

export function getTestDemo(Comp) {
    return forwardRef((props = {}, ref) => {
        const [visible, setVisible] = useState(false);
        const compRef = useRef(null);

        useImperativeHandle(ref, () => compRef.current);

        return (
            <>
                <Button id="demo-btn" onClick={() => setVisible(true)} type="primary">
                    Open
                </Button>
                <Comp
                    visible={visible}
                    close={() => setVisible(false)}
                    ref={compRef}
                    {...props}
                >
                    Content
                </Comp>
            </>
        );
    });
}

export function openMasking(wrapper, Comp) {
    wrapper.find('#demo-btn').simulate('click');
    expect(wrapper.find(Comp).props().visible).toBe(true);
    act(() => {
        jest.advanceTimersByTime(800);
    });
    wrapper.update();
}

export function testMaskingCase(compName, Comp, prefix, contentClass = `${prefix}-content`, maskClass = `${prefix}-mask`) {
    it(`should open ${compName} correctly`, () => {
        const onOpen = jest.fn();
        const onClose = jest.fn();
        const onMaskClick = jest.fn();
        const TestDemo = getTestDemo(Comp);
        const wrapper = mount(<TestDemo onOpen={onOpen} onClose={onClose} onMaskClick={onMaskClick} maskClosable={false} />);
        expect(document.querySelectorAll(`.${prefix}`)).toHaveLength(0);
        openMasking(wrapper, Comp);
        expect(document.querySelectorAll(`.${prefix}`)).toHaveLength(1);
        expect(onOpen.mock.calls).toHaveLength(1);
        expect(wrapper.find(`.${maskClass}`)).toHaveLength(1);
        // simulate close masking
        wrapper.find(`.${maskClass}`).simulate('click');
        expect(onMaskClick.mock.calls).toHaveLength(1);
        expect(document.querySelectorAll(`.${prefix}`)).toHaveLength(1);
        expect(wrapper.find(Comp).props().visible).toBe(true);
        wrapper.setProps({ maskClosable: true });
        wrapper.find(`.${maskClass}`).simulate('click');
        expect(wrapper.find(Comp).props().visible).toBe(false);
        act(() => {
            jest.advanceTimersByTime(800);
        });
        wrapper.update();
        expect(onClose.mock.calls).toHaveLength(1);
        expect(onClose.mock.calls[0]).toEqual(['mask']);
        expect(document.querySelectorAll(`.${prefix}`)).toHaveLength(0);
    });

    it('should keep dom when invisible and `mountOnEnter=false` and `unmountOnExit=false`', () => {
        const TestDemo = getTestDemo(Comp);
        const wrapper = mount(<TestDemo mountOnEnter={false} unmountOnExit={false} />);
        expect(document.querySelectorAll(`.${prefix}`)).toHaveLength(1);
        expect(wrapper.find(`.${maskClass}`)).toHaveLength(1);
        expect(wrapper.find(`.${contentClass}`)).toHaveLength(1);
        expect(wrapper.find(`.${maskClass}`).hasClass('pre-mount')).toBe(true);
        expect(wrapper.find(`.${contentClass}`).hasClass('pre-mount')).toBe(true);
        openMasking(wrapper, Comp);
        expect(document.querySelectorAll(`.${prefix}`)).toHaveLength(1);
        expect(wrapper.find(`.${maskClass}`).hasClass('pre-mount')).toBe(false);
    });

    it('should support `open` and `setCloseScene`', () => {
        const onClose = jest.fn();
        const ref = React.createRef();
        window.maskingInstance = Comp.open({
            ref,
            onClose,
            maskTransitionTimeout: 1000,
            className: 'demo-global',
            children: 'Content',
        });
        act(() => {
            jest.advanceTimersByTime(1100);
        });
        expect(document.querySelectorAll('.demo-global')).toHaveLength(1);
        expect(typeof window.maskingInstance.close).toBe('function');
        expect(typeof window.maskingInstance.update).toBe('function');
        const { setCloseScene } = ref.current;
        expect(typeof setCloseScene).toBe('function');
        setCloseScene('global');
        window.maskingInstance.close();
        act(() => {
            jest.advanceTimersByTime(1100);
        });
        expect(onClose.mock.calls).toHaveLength(1);
        expect(onClose.mock.calls[0]).toEqual(['global']);
    });
}
