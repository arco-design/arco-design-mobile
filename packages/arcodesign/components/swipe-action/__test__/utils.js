import { act, fireEvent } from '@testing-library/react';
import {
    createMoveTouchEventObject,
    createStartTouchEventObject,
} from '../../../tests/helpers/mockEvent';

export function mockSwipeStart(comp, clientXMap) {
    const { touchstart, touchmove } = clientXMap;
    act(() => {
        fireEvent.touchStart(
            comp,
            createStartTouchEventObject({ x: touchstart, px: touchstart, y: 0 }),
        );
        fireEvent.touchMove(
            comp,
            createMoveTouchEventObject({ x: touchmove, px: touchmove, y: 0 }),
        );
    });
}
export function mockSwipeEnd(comp, touchend) {
    fireEvent.touchEnd(comp, createMoveTouchEventObject({ x: touchend, px: touchend, y: 0 }));
    act(() => {
        jest.advanceTimersByTime(600);
    });
}

export function mockSwipe(comp, clientXMap) {
    const { touchend } = clientXMap;
    mockSwipeStart(comp, clientXMap);
    mockSwipeEnd(comp, touchend);
}
