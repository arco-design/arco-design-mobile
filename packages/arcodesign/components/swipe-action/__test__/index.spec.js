import React, { createRef } from 'react';
import { act, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import { defaultContext } from '../../context-provider';
import SwipeAction from '..';
import { mockSwipe } from './utils';

demoTest('swipe-action');

mountTest(SwipeAction, 'SwipeAction');

const prefix = `${defaultContext.prefixCls}-swipe-action`;

const props = {
    rightActions: [
        {
            text: '置顶',
            style: {
                background: '#C9CDD4',
            },
        },
        {
            text: '删除',
            style: {
                background: '#F53F3F',
            },
        },
        {
            text: '点赞',
            style: {
                background: '#C9CDD4',
            },
        },
    ],
    leftActions: [
        {
            text: '置顶',
            style: {
                background: '#C9CDD4',
            },
            onClick: () => false,
        },
        {
            text: '删除',
            style: {
                background: '#F53F3F',
            },
        },
    ],
};

const contentStyle = {
    width: '100%',
    height: '50px',
};
function ContentDiv() {
    return <div style={contentStyle} />;
}

describe('SwipeAction', () => {
    beforeEach(() => {
        jest.useFakeTimers();
    });

    afterEach(() => {
        jest.useRealTimers();
    });

    it('should render correctly when set swipe-action and children', () => {
        const { container } = render(
            <SwipeAction {...props}>
                <ContentDiv />
            </SwipeAction>,
        );
        expect(container.querySelectorAll(`.${prefix}`).length).toBe(1);
        expect(container.querySelectorAll(`.${prefix}-menu`).length).toBe(2);
        expect(container.querySelectorAll(`.${prefix}-menu-action-left`).length).toBe(2);
        expect(container.querySelectorAll(`.${prefix}-menu-action-right`).length).toBe(3);
    });

    it('should should support swipe-action to onClose and onOpen', async () => {
        const mockOpenFn = jest.fn();
        const mockCloseFn = jest.fn();
        const ref = createRef();
        render(
            <SwipeAction {...props} ref={ref} onOpen={mockOpenFn} onClose={mockCloseFn}>
                <ContentDiv />
            </SwipeAction>,
        );
        const wrapper = ref.current;
        wrapper.open('left');
        await act(async () => {
            await jest.advanceTimersByTime(500);
        });
        expect(mockOpenFn).toBeCalled();
        wrapper.close();
        await act(async () => {
            await jest.advanceTimersByTime(500);
        });
        expect(mockCloseFn).toBeCalled();
    });

    it('should allow to drag to show menu', async () => {
        const mockOpenFn = jest.fn();
        const originalGetBoundingClientRect = Element.prototype.getBoundingClientRect;
        Element.prototype.getBoundingClientRect = () => ({
            width: 64,
        });
        const { container } = render(
            <SwipeAction {...props} onOpen={mockOpenFn}>
                <ContentDiv />
            </SwipeAction>,
        );
        const comp = container.querySelector(`.${prefix}`);
        mockSwipe(comp, { touchstart: 100, touchmove: 200, touchend: 400 });
        await act(async () => {
            await jest.advanceTimersByTime(500);
        });
        expect(mockOpenFn).toBeCalled();
        Element.prototype.getBoundingClientRect = originalGetBoundingClientRect;
    });

    it('click document the swipe action should close', async () => {
        const ref = createRef();
        const mockFn = jest.fn();
        const { container } = render(
            <SwipeAction closeOnTouchOutside ref={ref} {...props} onClose={mockFn}>
                <ContentDiv />
            </SwipeAction>,
        );
        const SwipeElement = ref.current;
        const event = new Event('touchstart');

        act(() => {
            SwipeElement.open('right');
            document.dispatchEvent(event);
        });
        await act(async () => {
            await jest.advanceTimersByTime(500);
        });
        expect(mockFn).toBeCalled();
        act(() => {
            SwipeElement.open('right');
        });
        userEvent.click(container.querySelector(`.${prefix}-menu-action-left`));
        await act(async () => {
            await jest.advanceTimersByTime(500);
        });
        expect(mockFn).toBeCalledTimes(2);
    });
});
