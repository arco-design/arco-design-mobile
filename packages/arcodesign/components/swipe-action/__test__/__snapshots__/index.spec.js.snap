// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`swipe-action demo test swipe-action demo: custom-color.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body"
    >
      <div
        class="arco-swipe-action action-close"
        style="transition-duration: 300ms; transform: translateX(0px);"
      >
        <div
          class="arco-swipe-action-menu-left arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        />
        <div
          class="arco-swipe-action-content"
        >
          <div
            class="arco-cell all-border-box bordered"
          >
            <div
              class="arco-cell-inner"
            >
              <div
                class="cell-label"
              >
                <div
                  class="cell-title"
                >
                  自定义颜色
                </div>
              </div>
              <div
                class="cell-content has-label"
              />
            </div>
          </div>
        </div>
        <div
          class="arco-swipe-action-menu-right arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        >
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 1; left: 0%; background: rgb(134, 144, 156);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                取消置顶
              </div>
            </div>
          </div>
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 2; left: 0%; background: rgb(245, 63, 63);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                删除
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`swipe-action demo test swipe-action demo: custom-style.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body"
    >
      <div
        class="arco-swipe-action action-close"
        style="transition-duration: 300ms; transform: translateX(0px);"
      >
        <div
          class="arco-swipe-action-menu-left arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        />
        <div
          class="arco-swipe-action-content"
        >
          <div
            class="arco-cell all-border-box bordered"
          >
            <div
              class="arco-cell-inner"
            >
              <div
                class="cell-label"
              >
                <div
                  class="cell-title"
                >
                  自定义样式
                </div>
              </div>
              <div
                class="cell-content has-label"
              />
            </div>
          </div>
        </div>
        <div
          class="arco-swipe-action-menu-right arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        >
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 1; left: 0%; background: rgb(255, 180, 0);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-icon"
              >
                <svg
                  class="arco-icon arco-icon-star "
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M268.437 941.29a35.563 35.563 0 01-50.773-35.733l25.899-248.49a35.563 35.563 0 00-8.896-27.456L66.176 442.155a35.563 35.563 0 0119.008-58.539l246.507-52.608a35.563 35.563 0 0023.36-16.96L481.237 95.872a35.563 35.563 0 0161.568 0l126.187 218.176a35.525 35.525 0 0023.36 16.96l246.507 52.608a35.563 35.563 0 0119.008 58.539L789.376 629.61a35.533 35.533 0 00-8.917 27.456l25.898 248.49a35.563 35.563 0 01-50.773 35.734l-228.16-109.739a35.563 35.563 0 00-30.805 0l-228.16 109.739zm191.147-186.645a120.896 120.896 0 01104.81 0l147.84 71.104-16.682-159.829a120.896 120.896 0 0130.336-93.355L836.864 449.11l-162.347-34.645a120.896 120.896 0 01-79.402-57.685L512 213.099l-83.115 143.68a120.896 120.896 0 01-79.402 57.685l-162.347 34.645 110.976 123.456a120.896 120.896 0 0130.336 93.355l-16.683 159.83 147.84-71.105z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                收藏
              </div>
            </div>
          </div>
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 2; left: 0%; background: rgb(245, 63, 63);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-icon"
              >
                <svg
                  class="arco-icon arco-icon-delete "
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M640 85.333A42.667 42.667 0 01682.667 128l-.022 42.645 228.694.022c9.493 0 12.928.981 16.426 2.858a19.41 19.41 0 018.043 8.064c1.877 3.478 2.859 6.912 2.859 16.427v30.635c0 9.514-.982 12.949-2.859 16.426a19.392 19.392 0 01-8.064 8.064c-3.477 1.878-6.912 2.859-16.427 2.859l-57.984-.021V896a42.667 42.667 0 01-42.666 42.667H213.333A42.667 42.667 0 01170.667 896l-.022-640.021-57.962.021c-9.515 0-12.95-.981-16.427-2.859a19.392 19.392 0 01-8.064-8.064c-1.877-3.477-2.859-6.912-2.859-16.426v-30.635c0-9.515.982-12.95 2.859-16.427a19.392 19.392 0 018.064-8.064c3.477-1.877 6.912-2.858 16.427-2.858l228.629-.022.021-42.645A42.667 42.667 0 01384 85.333h256zM768 256H256v597.333h512V256zM448 384a21.333 21.333 0 0121.333 21.333V704A21.333 21.333 0 01448 725.333h-42.667A21.333 21.333 0 01384 704V405.333A21.333 21.333 0 01405.333 384H448zm170.667 0A21.333 21.333 0 01640 405.333V704a21.333 21.333 0 01-21.333 21.333H576A21.333 21.333 0 01554.667 704V405.333A21.333 21.333 0 01576 384h42.667z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                删除
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`swipe-action demo test swipe-action demo: left-right.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body"
    >
      <div
        class="arco-swipe-action action-close"
        style="transition-duration: 300ms; transform: translateX(0px);"
      >
        <div
          class="arco-swipe-action-menu-left arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        >
          <div
            class="arco-swipe-action-menu-action-left arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 2; right: 0%; background: rgb(201, 205, 212);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                置顶
              </div>
            </div>
          </div>
          <div
            class="arco-swipe-action-menu-action-left arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 1; right: 0%; background: rgb(245, 63, 63);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                删除
              </div>
            </div>
          </div>
        </div>
        <div
          class="arco-swipe-action-content"
        >
          <div
            class="arco-cell all-border-box bordered"
          >
            <div
              class="arco-cell-inner"
            >
              <div
                class="cell-label"
              >
                <div
                  class="cell-title"
                >
                  左右滑动
                </div>
              </div>
              <div
                class="cell-content has-label"
              />
            </div>
          </div>
        </div>
        <div
          class="arco-swipe-action-menu-right arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        >
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 1; left: 0%; background: rgb(201, 205, 212);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                取消置顶
              </div>
            </div>
          </div>
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 2; left: 0%; background: rgb(245, 63, 63);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                删除
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`swipe-action demo test swipe-action demo: multiple-lines.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body"
    >
      <div
        class="arco-swipe-action action-close"
        style="transition-duration: 300ms; transform: translateX(0px);"
      >
        <div
          class="arco-swipe-action-menu-left arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        >
          <div
            class="arco-swipe-action-menu-action-left arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 2; right: 0%; background: rgb(201, 205, 212);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                置顶
              </div>
            </div>
          </div>
          <div
            class="arco-swipe-action-menu-action-left arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 1; right: 0%; background: rgb(245, 63, 63);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                删除
              </div>
            </div>
          </div>
        </div>
        <div
          class="arco-swipe-action-content"
        >
          <div
            class="arco-cell all-border-box bordered"
          >
            <div
              class="arco-cell-inner has-desc"
            >
              <div
                class="cell-label"
              >
                <div
                  class="cell-title"
                >
                  多行数值
                </div>
                <div
                  class="cell-desc"
                >
                  辅助文字
                </div>
              </div>
              <div
                class="cell-content has-label"
              />
            </div>
          </div>
        </div>
        <div
          class="arco-swipe-action-menu-right arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`swipe-action demo test swipe-action demo: secondary-confirmation.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body"
    >
      <div
        class="arco-swipe-action action-close"
        style="transition-duration: 300ms; transform: translateX(0px);"
      >
        <div
          class="arco-swipe-action-menu-left arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        />
        <div
          class="arco-swipe-action-content"
        >
          <div
            class="arco-cell all-border-box bordered"
          >
            <div
              class="arco-cell-inner"
            >
              <div
                class="cell-label"
              >
                <div
                  class="cell-title"
                >
                  二次确认
                </div>
              </div>
              <div
                class="cell-content has-label"
              />
            </div>
          </div>
        </div>
        <div
          class="arco-swipe-action-menu-right arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        >
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 1; left: 0%; background: rgb(201, 205, 212);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                取消置顶
              </div>
            </div>
          </div>
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 2; left: 0%; background: rgb(245, 63, 63);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                删除
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`swipe-action demo test swipe-action demo: single-line.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body"
    >
      <div
        class="arco-swipe-action action-close"
        style="transition-duration: 300ms; transform: translateX(0px);"
      >
        <div
          class="arco-swipe-action-menu-left arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        />
        <div
          class="arco-swipe-action-content"
        >
          <div
            class="arco-cell all-border-box bordered"
          >
            <div
              class="arco-cell-inner"
            >
              <div
                class="cell-label"
              >
                <div
                  class="cell-title"
                >
                  单行数值
                </div>
              </div>
              <div
                class="cell-content has-label"
              />
            </div>
          </div>
        </div>
        <div
          class="arco-swipe-action-menu-right arco-swipe-action-menu action-close"
          style="width: 0px; transition-duration: 300ms;"
        >
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 1; left: 0%; background: rgb(201, 205, 212);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                置顶
              </div>
            </div>
          </div>
          <div
            class="arco-swipe-action-menu-action-right arco-swipe-action-menu-action-info-container open-style-layer"
            style="z-index: 2; left: 0%; background: rgb(245, 63, 63);"
          >
            <div
              class="arco-swipe-action-menu-action-info"
            >
              <div
                class="arco-swipe-action-menu-action-info-text"
              >
                删除
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
